tendrl/commons/flows/create_cluster/__init__.py

```diff
@@ -232,3 +232,10 @@ def run(self):
         finally:
             # releasing nodes if any exception came
             create_cluster_utils.release_node_lock(self.parameters)
+
+            # Persist the cluster's extra details
+            NS.tendrl.objects.Cluster(
```

Comment:
May be this could be done just after releasing the node locks, as that's the place where create is done at least.